src/module/__tests__/helpers-dice.test.ts

```diff
@@ -0,0 +1,484 @@
+    it("Natural 20 total always succeed", () => {
+      const roll = createMockRoll(20);
+      expect(OseDice.attackIsSuccess(roll, 100, 10)).equal(true);
+    });
+    it("Natural 20+ total always succeed", () => {
+      const roll = createMockRoll(21);
+      expect(OseDice.attackIsSuccess(roll, 100, 10)).equal(true);
+    });
+    it("Natural 20 terms always succeed", () => {
+      const roll = createMockRoll(5, [20]);
+      expect(OseDice.attackIsSuccess(roll, 100, 10)).equal(true);
+    });
+    it("Roll + ac equal thac0 is successful", () => {
+      const roll = createMockRoll(10);
+      const thac0 = 20;
+      const ac = 10;
+      expect(OseDice.attackIsSuccess(roll, thac0, ac)).equal(true);
+    });
+    it("Roll + ac above thac0 is successful", () => {
+      const roll = createMockRoll(10);
+      const thac0 = 19;
+      const ac = 10;
+      expect(OseDice.attackIsSuccess(roll, thac0, ac)).equal(true);
+    });
+    it("Roll + ac under thac0 is unsuccessful", () => {
+      const roll = createMockRoll(10);
+      const thac0 = 21;
+      const ac = 10;
+      expect(OseDice.attackIsSuccess(roll, thac0, ac)).equal(false);
+    });
+  });
+  describe("digestAttackResult(data, roll)", () => {
+    const data = {
+      roll: {
+        thac0: 15,
+        target: {
+          actor: {
+            system: {
+              ac: { value: 0 },
+              aac: { value: 9 },
+            },
+          },
+        },
+      },
+    };
+    describe("Ascending AC", () => {
+      it("Natural 1 terms is unsuccesful", async () => {
+        await game.settings.set(game.system.id, "ascendingAC", true);
+        expect(game.settings.get(game.system.id, "ascendingAC")).equal(true);
+        const roll = createMockRoll(1);
+        expect(OseDice.digestAttackResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestAttackResult(data, roll).isFailure).equal(true);
+      });
+      it("Lower than target AC is unsuccesful", () => {
+        const roll = createMockRoll(8);
+        expect(OseDice.digestAttackResult(data, roll).isSuccess).equal(false);
+        expect(OseDice.digestAttackResult(data, roll).isFailure).equal(true);
+      });
+      it("Equal than target AC is succesful", () => {
+        const roll = createMockRoll(9);
+        expect(OseDice.digestAttackResult(data, roll).isSuccess).equal(true);
+        expect(OseDice.digestAttackResult(data, roll).isFailure).equal(false);
+      });
+      it("Higher than target AC is succesful", () => {
+        const roll = createMockRoll(11);
+        expect(OseDice.digestAttackResult(data, roll).isSuccess).equal(true);
+        expect(OseDice.digestAttackResult(data, roll).isFailure).equal(false);
+      });
+      it("Natural 20 is succesful", () => {
+        const roll = createMockRoll(20);
+        expect(OseDice.digestAttackResult(data, roll).isSuccess).equal(true);
+        expect(OseDice.digestAttackResult(data, roll).isFailure).equal(false);
+      });
+    });
```

Comment:
Need another `createMockRoll(1, [20])` here

Reply:
Fixed, see previous comment on the Natural 1.